Lib/Schema/Schema.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * Schema class file.
+ *
+ * @package Freyja\Database\Schema
+ * @copyright 2016 SqueezyWeb
+ * @since 0.1.0
+ */
+
+namespace Freyja\Database\Schema;
+
+use Freyja\Database\Driver;
+use Freyja\Database\Table;
+use Freyja\Database\Database;
+use Symfony\Component\Yaml\Yaml;
+use Freyja\Exceptions\InvalidArgumentException as InvArgExcp;
+
+/**
+ * Schema class.
+ *
+ * @package Freyja\Database\Schema
+ * @author Gianluca Merlo <gianluca@squeezyweb.com>
+ * @since 0.1.0
+ * @version 1.0.0
+ */
+class Schema {
+  /**
+   * Database schema.
+   *
+   * @todo: put an example.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $schema = array();
+
+  /**
+   * Database.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var Freyja\Database\Database
+   */
+  private $database;
+
+  /**
+   * Configuration file path.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $filename;
+
+  /**
+   * Class constructor.
+   *
+   * The Database argument MUST be connected before passing it to the
+   * constructor.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param Freyja\Database\Database $database
+   */
+  public function __construct(Database $database) {
+    $this->database = $database;
+
+    $this->filename = getcwd().'/db/Schema.yml';
+    if (file_exists($this->filename)) {
+      $schema = Yaml::parse(file_get_contents($this->filename));
+      $db_name = $schema[$database->getName()];
+      if (isset($db_name))
```

Comment:
You declare `$db_name` in the previous line, so it's always set.
Declare `$db_name` (or, better, `$db`) as follows:
```php
$db = isset($schema[$database->getName()]) ? $schema[$database->getName() : '';
```
Then you can check properly:
```php
if (!empty($db))
```